ext/gd/gd.c

```diff
@@ -4108,27 +4108,50 @@ static void _php_image_output_ctx(INTERNAL_FUNCTION_PARAMETERS, int image_type,
 	switch (image_type) {
 #ifdef HAVE_GD_JPG
 		case PHP_GDIMG_TYPE_JPG:
+			if (quality < -1 || quality > 100) {
+				zend_argument_value_error(3, "must be at between -1 and 100");
+				ctx->gd_free(ctx);
+				RETURN_THROWS();
+			}
 			gdImageJpegCtx(im, ctx, (int) quality);
 			break;
 #endif
 #ifdef HAVE_GD_WEBP
 		case PHP_GDIMG_TYPE_WEBP:
-			if (quality == -1) {
+			if (quality < -1) {
+				zend_argument_value_error(3, "must be at least -1");
+				ctx->gd_free(ctx);
+				RETURN_THROWS();
+			} else if (quality == -1) {
```

Comment:
This "else if" branch is actually unnecessary:

https://github.com/php/php-src/blob/c7366cf6a5c32f90aff255bee613574c7b68bc69/ext/gd/libgd/gd_webp.c#L120-L122

Reply:
yes saw that but was wondering if this is a "recent" addition was wondering why it is in place here then.

Reply:
It seems that the code previously used virtual functions so that might've been harder to see what the possible call targets are and therefore harder to see what's necessary and not. I'm fine with keeping this of course.
It's been in the library part since 2015: 00ba7e6215dead29b6a7c2aaae7a217a39513de0

Reply:
I see. happy to remove anyway :)